lib/pure/parsecfg.nim

```diff
@@ -19,37 +19,33 @@
 ##     :literal:
 ##
 ## Here is an example of how to use the configuration file parser:
-##
-## .. code-block:: nim
-##
-##    import std/[os, parsecfg, strutils, streams]
-##
-##    var f = newFileStream(paramStr(1), fmRead)
-##    if f != nil:
-##      var p: CfgParser
-##      open(p, f, paramStr(1))
-##      while true:
-##        var e = next(p)
-##        case e.kind
-##        of cfgEof: break
-##        of cfgSectionStart:   ## a `[section]` has been parsed
-##          echo("new section: " & e.section)
-##        of cfgKeyValuePair:
-##          echo("key-value-pair: " & e.key & ": " & e.value)
-##        of cfgOption:
-##          echo("command: " & e.key & ": " & e.value)
-##        of cfgError:
-##          echo(e.msg)
-##      close(p)
-##    else:
-##      echo("cannot open: " & paramStr(1))
-##
-##
-## Examples
-## ========
-##
+runnableExamples("-r:off"):
+  import std/[os, strutils, streams]
+
+  var f = newFileStream(paramStr(1), fmRead)
```

Comment:
pre-existing but `paramStr` has nothing to do with parsecfg though; how about change it to:

```nim
let configFile = "example.conf"
 # or even: const configFile {.strdefine.} = ""
var f = newFileStream(configFile, fmRead)
```
replace
`if f != nil`
by:
`assert f != nil, "cannot open " & configFile`

Reply:
well

Reply:
fixed